src/pydicom/valuerep.py

```diff
@@ -1108,7 +1108,7 @@ def __str__(self) -> str:
         return repr(self)[1:-1]
 
     def __repr__(self) -> str:
-        if self.auto_format and hasattr(self, "original_string"):
```

Comment:
I actually don't understand in what situations `autoformat` is needed.

Reply:
It's a bit of an odd thing, you'd only be able to use it if creating your own values, in which case you'd presumably just format them correctly beforehand?

Reply:
Yes, that's what I've thought, too... Maybe @darcymason knows more?

Reply:
It came from #1264/#1334, just a shorthand way to allow users to fix DS problems automatically.